pkg/resourceinterpreter/configurableinterpreter/interpreters.go

```diff
@@ -0,0 +1,212 @@
+package configurableinterpreter
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/resourceinterpreter/configurableinterpreter/engine"
+)
+
+type Interpreter interface {
+	IsEnabled(operation configv1alpha1.InterpreterOperation) bool
+
+	Retain(desired *unstructured.Unstructured, observed *unstructured.Unstructured) (retained *unstructured.Unstructured, enabled bool, err error)
+
+	GetReplicas(object *unstructured.Unstructured) (replicas int32, require *workv1alpha2.ReplicaRequirements, enabled bool, err error)
+
+	ReviseReplica(object *unstructured.Unstructured, replica int64) (obj *unstructured.Unstructured, enabled bool, err error)
+
+	AggregateStatus(object *unstructured.Unstructured, aggregatedStatusItems []workv1alpha2.AggregatedStatusItem) (obj *unstructured.Unstructured, enabled bool, err error)
+
+	GetDependencies(object *unstructured.Unstructured) (dependencies []configv1alpha1.DependentObjectReference, enabled bool, err error)
+
+	ReflectStatus(object *unstructured.Unstructured) (status *runtime.RawExtension, enabled bool, err error)
+
+	InterpretHealth(object *unstructured.Unstructured) (healthy bool, enabled bool, err error)
+}
+
+type InterpreterManager interface {
+	LoadConfig(customizations configv1alpha1.CustomizationRules) error
+
+	Interpreter
+}
+
+func NewInterpreterManager() InterpreterManager {
+	return InterpreterConfigs{
+		&LuaScriptInterpreter{},
+	}
+}
+
+type InterpreterConfigs []InterpreterManager
+
+var _ Interpreter = (*InterpreterConfigs)(nil)
```

Comment:
Hard to say, just feeling weird.
Can we use another struct to get rid of the slice here?